test/Generics/redundant_constraints.swift

```diff
@@ -0,0 +1,144 @@
+
+struct S10<T where T : Any> where T : Any {}
+// expected-warning@-1 {{redundant conformance constraint 'T': 'Any'}} {{14-27=}}
+// expected-note@-2 {{all types implicitly conform to 'Any'}}
+// expected-warning@-3 {{redundant conformance constraint 'T': 'Any'}} {{29-43=}}
+// expected-note@-4 {{all types implicitly conform to 'Any'}}
+// expected-warning@-5 {{is deprecated}}
+
+struct S11<T where T : Any, T : P1> where T : Any {}
+// expected-warning@-1 {{redundant conformance constraint 'T': 'Any'}} {{20-29=}}
+// expected-note@-2 {{all types implicitly conform to 'Any'}}
+// expected-warning@-3 {{redundant conformance constraint 'T': 'Any'}} {{37-51=}}
+// expected-note@-4 {{all types implicitly conform to 'Any'}}
+// expected-warning@-5 {{is deprecated}}
+
+struct S12<T where T : P1, T : Any> where T : Any {}
+// expected-warning@-1 {{redundant conformance constraint 'T': 'Any'}} {{26-35=}}
+// expected-note@-2 {{all types implicitly conform to 'Any'}}
+// expected-warning@-3 {{redundant conformance constraint 'T': 'Any'}} {{37-51=}}
+// expected-note@-4 {{all types implicitly conform to 'Any'}}
+// expected-warning@-5 {{is deprecated}}
+
+struct S13<T where T : Any, T : P1> where T : Any, T : P2 {}
+// expected-warning@-1 {{redundant conformance constraint 'T': 'Any'}} {{20-29=}}
+// expected-note@-2 {{all types implicitly conform to 'Any'}}
+// expected-warning@-3 {{redundant conformance constraint 'T': 'Any'}} {{43-52=}}
+// expected-note@-4 {{all types implicitly conform to 'Any'}}
+// expected-warning@-5 {{is deprecated}}
+
+struct S14<T where T : P1, T : Any> where T : P2, T : Any {}
+// expected-warning@-1 {{redundant conformance constraint 'T': 'Any'}} {{26-35=}}
+// expected-note@-2 {{all types implicitly conform to 'Any'}}
+// expected-warning@-3 {{redundant conformance constraint 'T': 'Any'}} {{49-58=}}
+// expected-note@-4 {{all types implicitly conform to 'Any'}}
+// expected-warning@-5 {{is deprecated}}
+
+struct S15<T, U> {}
+extension S15 where T : Any {} // expected-warning {{redundant conformance constraint 'T': 'Any'}} {{15-29=}}
+// expected-note@-1 {{all types implicitly conform to 'Any'}}
+
+extension S15 where T : Any, T : P1 {} // expected-warning {{redundant conformance constraint 'T': 'Any'}} {{21-30=}}
+// expected-note@-1 {{all types implicitly conform to 'Any'}}
+
+extension S15 where T : Any, U : Any {}
+// expected-warning@-1 {{redundant conformance constraint 'T': 'Any'}} {{21-30=}}
+// expected-note@-2 {{all types implicitly conform to 'Any'}}
+// expected-warning@-3 {{redundant conformance constraint 'U': 'Any'}} {{28-37=}}
+// expected-note@-4 {{all types implicitly conform to 'Any'}}
+
+
+protocol P3 {
+  // expected-warning@-1 {{redundant conformance constraint 'Self.X1': 'Any'}}
+  // expected-note@-2 {{all types implicitly conform to 'Any'}}
+  // expected-warning@-3 {{redundant conformance constraint 'Self.X2': 'Any'}} {{21-35=}}
+  // expected-note@-4 {{all types implicitly conform to 'Any'}}
+
```

Comment:
Diagnosing on the protocol declaration location for constraints relating to associated types is what we already do for other diagnostics in GSB, so I haven't changed the behaviour here.

Is this intentional?